tests/basic/basic_test.go

```diff
@@ -572,6 +574,70 @@ var _ = Describe("{CordonStorageNodesDeployDestroy}", func() {
 	})
 })
 
+var _ = Describe("{SecretsVaultFunctional}", func() {
+	var testrailID, runID int
+	var contexts []*scheduler.Context
+
+	const (
+		secretsProvider       = "vault"
+		portworxContainerName = "portworx"
+	)
+
+	BeforeEach(func() {
+		runID = testrailuttils.AddRunsToMilestone(testrailID)
+		isOpBased, _ := Inst().V.IsOperatorBasedInstall()
+		if !isOpBased {
+			k8sApps := apps.Instance()
+			deployments, err := k8sApps.ListDaemonSets("kube-system", metav1.ListOptions{
+				LabelSelector: "name=portworx",
+			})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(deployments)).NotTo(Equal(0))
+			Expect(deployments[0].Spec.Template.Spec.Containers).NotTo(BeEmpty())
+			usingVault := false
+			for _, container := range deployments[0].Spec.Template.Spec.Containers {
+				if container.Name == portworxContainerName {
+					for _, arg := range container.Args {
+						if arg == secretsProvider {
+							usingVault = true
+						}
+					}
+				}
+			}
+			if !usingVault {
+				Skip(fmt.Sprintf("Skip test for not using %s", secretsProvider))
+			}
+		} else {
+			spec, err := Inst().V.GetStorageCluster()
+			Expect(err).ToNot(HaveOccurred())
+			if *spec.Spec.SecretsProvider != secretsProvider {
+				Skip(fmt.Sprintf("Skip test for not using %s", secretsProvider))
+			}
+		}
+	})
+
+	var _ = Describe("{RunSecretsLogin}", func() {
+		var testrailID = 82774
+		// testrailID corresponds to: https://portworx.testrail.net/index.php?/cases/view/82774
+		var runID int
+
+		It("has to runs secrets login for vault", func() {
+			contexts = make([]*scheduler.Context, 0)
+			n := node.GetWorkerNodes()[0]
+			err := Inst().V.RunSecretsLogin(n, secretsProvider)
+			Expect(err).ToNot(HaveOccurred())
+
+		})
+		JustAfterEach(func() {
+			AfterEachTest(contexts, testrailID, runID)
```

Comment:
AfterEachTest is called in the AfterEach() block as well. Probably we should call it only once per "It" test.

Reply:
updated.